src/savegame.cpp

```diff
@@ -437,6 +437,11 @@ bool savegame::check_filename(const std::string& filename)
 	if(filesystem::is_compressed_file(filename)) {
 		gui2::show_error_message(_("Save names should not end on '.gz' or '.bz2'. Please remove the extension."));
 		return false;
+	} else if(!filesystem::is_legal_user_file_name(filename)) {
+		// This message is not all-inclusive. This is on purpose. Few people
+		// need to know about DOS device names or the 255 character limit.
+		gui2::show_error_message(_("Save names may not end with a dot, or contain two dots or any of the following characters:\n    \" * / : < > ? \\ | ~"));
```

Comment:
It looks good to me.  Personally I usually include a comma between the final two parts of a list, but I don't know what Wesnoth's convention on that is.